packages/typespec-go/src/emitter.ts

```diff
@@ -63,9 +64,9 @@ export async function $onEmit(context: EmitContext<GoEmitterOptions>) {
     }
 
     // if we have a post-generation transforms file then "go generate" it
-    if (afterGenerateFileExists) {
+    if (goGenerateFileExists) {
       try {
-        execSync('go generate', { cwd: context.emitterOutputDir, encoding: 'ascii' });
+        execSync(`go generate ${goGenerateFile}`, { cwd: context.emitterOutputDir, encoding: 'ascii' });
       } catch (err) {
         context.program.reportDiagnostic({
```

Comment:
Does the error message return the `goGenerateFile` name? Putting the wrong file path could be a common stumbling block, and returning the file name would be useful information.

Reply:
This has been updated so that if `go-generate` is specified and the file isn't found an error is returned including the full path to the file (great idea).